app/src/main/java/fr/free/nrw/commons/mwapi/OkHttpJsonApiClient.java

```diff
@@ -330,7 +333,19 @@ public OkHttpJsonApiClient(OkHttpClient okHttpClient,
                 for (MwQueryPage page : pages) {
                     Media media = Media.from(page);
                     if (media != null) {
-                        mediaList.add(media);
+                        if(keyword.equals("Category:Uploaded_with_Mobile/Android")){
```

Comment:
Special case handling for `Category:Uploaded_with_Mobile/Android` category should not be the responsibility of `OkHttpJsonApiClient`. Please add this check in `CategoryImageController`. 

https://github.com/commons-app/apps-android-commons/blob/master/app/src/main/java/fr/free/nrw/commons/category/CategoryImageController.java